azurerm/internal/services/appservice/sourcecontrol/source_control_resource.go

```diff
@@ -0,0 +1,404 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/appservice/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/appservice/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+type AppServiceSourceControlResource struct{}
+
+type AppServiceSourceControlModel struct {
+	AppID                     string                      `tfschema:"app_id"`
+	SCMType                   string                      `tfschema:"scm_type"`
+	RepoURL                   string                      `tfschema:"repo_url"`
+	Branch                    string                      `tfschema:"branch"`
+	ManualIntegration         bool                        `tfschema:"manual_integration"`
+	UseMercurial              bool                        `tfschema:"use_mercurial"`
+	RollbackEnabled           bool                        `tfschema:"rollback_enabled"`
+	UsesGithubAction          bool                        `tfschema:"uses_github_action"`
+	GithubActionConfiguration []GithubActionConfiguration `tfschema:"github_action_configuration"`
+}
+
+var _ sdk.Resource = AppServiceSourceControlResource{}
+var _ sdk.ResourceWithUpdate = AppServiceSourceControlResource{}
+
+func (r AppServiceSourceControlResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"app_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.WebAppID,
+		},
+
+		"repo_url": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"scm_type": { // Note: this is largely determined by the service based on the provided `repo_url`. It is included here as it is required to set `LocalGit` and possibly for scenarios where the service cannot decode the URL correctly.
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			Computed: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(web.ScmTypeBitbucketGit),
+				string(web.ScmTypeBitbucketHg),
+				string(web.ScmTypeCodePlexGit),
+				string(web.ScmTypeCodePlexHg),
+				string(web.ScmTypeDropbox),
+				string(web.ScmTypeExternalGit),
+				string(web.ScmTypeExternalHg),
+				string(web.ScmTypeGitHub),
+				string(web.ScmTypeLocalGit),
+				string(web.ScmTypeOneDrive),
+				string(web.ScmTypeTfs),
+				string(web.ScmTypeVSO),
+			}, false),
+		},
+
+		"branch": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"manual_integration": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"uses_github_action": {
```

Comment:
tense compared to the rest of the provider
```suggestion
		"use_github_action": {
```
or
```suggestion
		"github_action_enabled": {
```

Reply:
I've moved this to a computed-only field as testing has shown it to need to be inferred by the presence of `github_action_configuration`, so have left the tense as-is.